invokeai/app/invocations/prompt.py

```diff
@@ -55,3 +57,41 @@ def invoke(self, context: InvocationContext) -> PromptCollectionOutput:
             prompts = generator.generate(self.prompt, num_images=self.max_prompts)
 
         return PromptCollectionOutput(prompt_collection=prompts, count=len(prompts))
+    
+
+class PromptsFromFileInvocation(BaseInvocation):
+    '''Loads prompts from a text file'''
+    # fmt: off
+    type: Literal['prompt_from_file'] = 'prompt_from_file'
+
+    # Inputs
+    file_path: str = Field(description="Path to prompt text file")
+    pre_prompt: Optional[str] = Field(description="String to prepend to each prompt")
+    post_prompt: Optional[str] = Field(description="String to append to each prompt")
+    start_line: int = Field(default=1, ge=1, description="Line in the file to start start from")
+    max_prompts: int = Field(default=1, ge=0, description="Max lines to read from file (0=all)")
+    #fmt: on
+
+    @validator("file_path")
+    def file_path_exists(cls, v):
```

Comment:
Technically, I think you'd end up with the same error state without the validator, but it doesn't hurt to have it.